main.tf

```diff
@@ -13,6 +13,39 @@ module "label" {
   tags       = "${var.tags}"
 }
 
+resource "aws_s3_bucket" "cache_bucket" {
+  bucket_prefix = "${local.b_name_hyphen}"
+  acl    = "private"
+  force_destroy = true
+  tags = "${module.label.tags}"
+
+  lifecycle_rule {
+    id      = "codebuildcache"
+    enabled = true
+
+    prefix  = "/"
+    tags = "${module.label.tags}"
+
+    expiration {
+      days = "${var.cache_timeout}"
+    }
+  }
+
+}
+
+locals {
+  b_name = "${module.label.id}"
+  b_name_hyphen = "${substr(join("-", split("_", lower(local.b_name))), 0, 20)}"
+  cache_def = {
+    "true" = [{
+              type     = "S3"
+              location = "${aws_s3_bucket.cache_bucket.bucket}"
+              }]
+    "false" = [{}]
+  }
+  cache = "${local.cache_def[var.cache]}"
```

Comment:
This is a nice trick.

Reply:
Thank's I made it up just to address backwards compatibility for your module :-D

I'll write up a little article on it.
As I think I can use the same trick to flexibly add additional environment flags.

Reply:
@osterman I wrote a pretty rough article here: https://www.linkedin.com/pulse/devops-how-flexibly-add-conditional-elements-terraform-jamie-nelson/